csm/conf/uds.py

```diff
@@ -0,0 +1,164 @@
+                        raise RuntimeError("Malformed UDS-related block in HAproxy config file")
+                    infile.seek(0)
+                    contents_before = infile.read(begin_uds_pos)
+                    if has_valid_block:
+                        after_end_uds_pos = end_uds_pos + len(cls.HAPROXY_END_UDS)
+                        infile.seek(after_end_uds_pos)
+                        contents_after = infile.read()
+                        has_repeated_block_delimiters = (cls.HAPROXY_BEGIN_UDS in contents_after
+                            or cls.HAPROXY_END_UDS in contents_after)
+                        if has_repeated_block_delimiters:
+                            raise RuntimeError(
+                                "Repeated UDS-related block delimiters in HAproxy config file. "
+                                "Please remove all repeated UDS-related entries (including dangling"
+                                " delimiters if there are any) from the file before proceeding to"
+                                " the config update.")
+                    else:
+                        contents_after = ''
+                    outfile.write(contents_before)
+                    if config is not None:
+                        outfile.write(
+                            f'{cls.HAPROXY_BEGIN_UDS}'
+                            f'{cls.HAPROXY_UDS_WARNING}'
+                            f'{config}'
+                            f'{cls.HAPROXY_END_UDS}'
+                        )
+                    outfile.write(contents_after)
+                    outfile.flush()
+                copyfile(outfile.name, infile.name)
+        finally:
+            os.umask(old_umask)
+
+    @classmethod
+    def update_haproxy_config(cls):
+        config = cls.generate_haproxy_config()
+        cls.write_haproxy_config(config)
+
+    @classmethod
+    def remove_haproxy_config(cls):
+        cls.write_haproxy_config(None)
+
+    @classmethod
+    def write_uds_config(cls, config):
+        uds_pwnam = getpwnam(cls.UDS_USERNAME)
+        uds_uid, uds_gid = uds_pwnam.pw_uid, uds_pwnam.pw_gid
+        old_umask = os.umask(0o077)
+        try:
+            Path(cls.UDS_CONFIG_DIR).mkdir(exist_ok=True)
+            os.chown(cls.UDS_CONFIG_DIR, uds_uid, uds_gid)
+            with open(cls.UDS_CONFIG_PATH, 'w') as f:
+                f.write(config)
+            os.chown(cls.UDS_CONFIG_PATH, uds_uid, uds_gid)
+        finally:
+            os.umask(old_umask)
+
+    @classmethod
+    def update_uds_config(cls):
+        config = cls.generate_uds_config()
+        cls.write_uds_config(f'{config}\n')
+
+    @classmethod
+    def remove_uds_config(cls):
+        rmtree(cls.UDS_CONFIG_DIR)
+
+    @classmethod
+    def apply(cls):
+        cls.update_haproxy_config()
+        cls.update_uds_config()
+
+    @classmethod
+    def delete(cls):
+        cls.remove_haproxy_config()
+        cls.remove_uds_config()
```

Comment:
@kupranay @ajaysrivas We can restart both `haproxy` and `uds` here using `systemctl` (if needed after `csm_setup reset`). Please advise.